### PR TITLE
[mds-cache] Clean up promise logic in hreads

### PR DESCRIPTION
**Pull request by avatarneil**
## PR Checklist

 - [ ] simple searchable title - `[mds-db] Add PG env var`, `[config] Fix eslint config`
 - [ ] briefly describe the changes in this PR
 - [ ] mark as draft if should not be merged
 - [ ] write tests for all new functionality

## Impacts
- [ ] Provider
- [x] Agency
- [x] Audit
- [ ] Policy
- [ ] Compliance
- [ ] Daily
- [ ] Native
- [ ] Policy Author

### CODE REVIEW

File: packages/mds-stream/index.ts [hunk 1]
```diff
@@ -93,7 +93,7 @@ declare module 'redis' {
 
   interface Multi {
     xadd: (...args: unknown[]) => string
-    execAsync: () => Promise<string[]>
+    execAsync: () => Promise<object[]>
```
Comment (avatarneil) [inline packages/mds-stream/index.ts]: This was most definitely not returning a `Promise<string[]>`, and previously wasn't caught due to the janky nested `Promise` implementation.

File: packages/mds-cache/index.ts [hunk 2]
```diff
@@ -256,33 +256,19 @@ async function hreads(
   // bleah
   const multi = (await getClient()).multi()
 
-  suffixes.map(suffix =>
-    ids.map(async id => {
-      await multi.hgetallAsync(decorateKey(`${prefix}:${id}:${suffix}`))
+  await suffixes.map(suffix =>
+    ids.map(id => {
+      return multi.hgetallAsync(decorateKey(`${prefix}:${id}:${suffix}`))
```
Comment (drtyh2o) [inline packages/mds-cache/index.ts]: Something still feels incorrect here. The inner map is going to return a `Promise[]` so then the outer map is going to return a `Promise[][]` which feels like an odd thing to `await`

File: packages/mds-cache/index.ts [hunk 3]
```diff
@@ -256,33 +256,19 @@ async function hreads(
...
-            if (flat) {
-              const flattened = { ...flat, [`${prefix}_id`]: ids[index % ids.length] }
-              return unflatten(flattened)
-            }
-            return unflatten(null)
-          })
-        )
-      }
-    })
+  const replies = await multi.execAsync()
```
Comment (BigIdeas) [inline packages/mds-cache/index.ts]: Audit will catch any error thrown here

File: packages/mds-cache/index.ts (hunk 2 above)
Comment (janedotx) [reply packages/mds-cache/index.ts]: Also shouldn't this be a `Promise.all`?

**Pull request merged**